x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -560,6 +561,37 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         }
         return false;
     }
+    
+    private static void checkHistogramInGrouping(LogicalPlan p, Set<Failure> localFailures) {
+        // check if the query has a grouping function (Histogram) but no GROUP BY
+        if (p instanceof Project) {
+            Project proj = (Project) p;
+            
+            proj.projections().forEach(e -> {
+                if (Functions.isGrouping(e) == false) {
+                    e.collectFirstChildren(c -> {
```

Comment:
`collectFirstChildren` uses a List to add the matches which we don't need in this impl.
Why don't you use `forEachDown`?

Reply:
Changed the approach and pushed a new commit.